### DISCUSSION TITLE
If `lastUpdated` enabled - date should also be exposed in sitemap and microdata (schema.org)

### DISCUSSION THREAD

**Discussion by stereobooster** [withastro/starlight#1848] [Feature Requests]
### What version of `starlight` are you using?

0.22.2

### What is your idea?

If `lastUpdated` enabled - it means that we have modification date for each (content) page. We can make use of it and expose it in:
- [sitemap](https://docs.astro.build/en/guides/integrations-guide/sitemap/) - `lastmod`. This should improve indexing of new content (theoretically - didn't test)
- [schema](https://schema.org/dateModified) - `dateModified`. This would be shown in search results for Google (probably other search engines too?)

### Why is this feature necessary?

see above

### Do you have examples of this feature in other projects?

n/a

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thanks for the suggestion @stereobooster! I think we could consider microformat markup for the date modified metadata on each page already? Would be happy to receive a PR for that.

Sitemap might be a bit trickier because of how Astro’s [sitemap integration](https://docs.astro.build/en/guides/integrations-guide/sitemap/) works, but I haven’t really explored the idea, so could also be something to think about.

I’d definitely recommend tackling the in-page metadata first though!

**Reply by Fryuni** [reply]
Having the last updated on the sitemap should be easily doable once the information is computed preemptively on the Starlight integration instead of on the renderers (part of the SSR support).

**Reply by stereobooster** [reply]
I wonder if we need full schema object (for example, https://schema.org/WebPage). Or adding microdata in the "last edit date" HTML is enough. There are validators https://developers.google.com/search/docs/appearance/structured-data need to experiment

**Reply by delucis** [reply]
Right, microdata in the HTML would be lovely if it can work!

**Reply by stereobooster** [reply]
Based on my very limited testing - Google wants full schema object. This is what I do now https://astro-digital-garden.stereobooster.com/recipes/seo-and-smo-meta-tags/#metadata. I may be wrong though

**Reply by delucis** [reply]
Yes, I also seem to be seeing that based on a quick search. That might be a bit tricky I guess because adding structured data would requires us making a few decisions about page type etc. which seems tricky to get right for everyone…

**Comment by patjakubik** [comment]
+1 this would be really nice to have
